### PR TITLE
Dark Mode Toggle

### PR DESCRIPTION
**Pull request by kre8sions**
#### Screens

Screen Shot 2020-09-08 at 3 56 19 PM

#### Breaking Changes

_Please reflect if your changes may break in some way (changes like removing/renaming props are examples of breaking changes)_

#### Checklist:

- [x] **DEPLOY** Please add the `Milano` label when you are ready for a review
- [x] **SCREENSHOT** Please add a screen shot or two
- [x] **SPECS** Please cover your changes with specs

### CODE REVIEW

**Comment by thestephenmarshall** [comment]
@kre8sions if this one is still WIP, you could convert to Draft.

File: app/helpers/playbook/pb_kit_helper.rb [hunk 1]
```diff
@@ -6,10 +6,11 @@
 
 module Playbook
   module PbKitHelper
+
     def pb_rails(kit, props: {}, &block)
       previous = prefix_partial_path_with_controller_namespace
       self.prefix_partial_path_with_controller_namespace = false
-      kit = build_view_model(kit.to_s, props, &block)
+      kit = build_view_model(kit.to_s, props.merge(dark: dark_mode), &block)
```
Comment (jasoncypret) [inline app/helpers/playbook/pb_kit_helper.rb:13]: Can we do a test in playgrounds to make sure this doesn't break?

File: app/pb_kits/playbook/config/_kit_example.html.erb [hunk 2]
```diff
@@ -1,9 +1,9 @@
 <%if !@kit_examples[:examples].nil? %>
   <% @kit_examples[:examples].each do |kit_example| %>
     <% kit_example.each do |key, item| %>
-      <div class="pb--doc <%=key.include?('_dark') ? 'dark_ui' : 'light_ui'%>">
+      <div class="pb--doc <%= @dark == "true" ? 'dark_ui' : 'light_ui'%>">
```
Comment (jasoncypret) [inline app/pb_kits/playbook/config/_kit_example.html.erb:4]: Can we replace with a pb_card?

File: app/pb_kits/playbook/packs/site_styles/docs/_kit_doc.scss [hunk 3]
```diff
@@ -143,11 +143,18 @@
   padding: 10px 30px 10px 30px;
   border: 1px solid $border-light;
 
+  &.dark {
+      background: #0a0527;
+      color: white;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/packs/site_styles/docs/_kit_doc.scss:148]: Won't this hide potential problems in our dark kits? I think perhaps adding this only to the markdown sections might be better?

File: app/pb_kits/playbook/packs/site_styles/docs/_markdown.scss [hunk 4]
```diff
@@ -2,6 +2,10 @@
 @import "../../../tokens/spacing";
 
 .markdown {
+  &.dark{
+    color: #fff;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/packs/site_styles/docs/_markdown.scss:6]:
```suggestion
    color: $white;
```

File: app/helpers/playbook/pb_doc_helper.rb [hunk 5]
```diff
@@ -45,6 +45,7 @@ def get_kit_footer(kit)
     end
 
     def pb_kit(kit: "", type: "rails", show_code: true)
+      # @dark = ENV["dark_mode"]
```
Comment (jasoncypret) [inline app/helpers/playbook/pb_doc_helper.rb:48]: Remove?

File: app/pb_kits/playbook/_playbook.scss [hunk 6]
```diff
@@ -15,6 +15,7 @@
 @import 'pb_currency/currency';
 @import 'pb_dashboard_value/dashboard_value';
 @import 'pb_date/date';
+@import 'pb_date_picker/date_picker';
```
Comment (jasoncypret) [inline app/pb_kits/playbook/_playbook.scss:18]: I think you need to merge from 6.6.0  or master

File: app/pb_kits/playbook/packs/site_styles/docs/_kit_doc.scss [hunk 7]
```diff
@@ -37,21 +36,17 @@
   }
 
   .pb--kit-example, .pb--kit-example-markdown {
+    &.dark{
+      color: $text_dk_light;
+      a{
+        color: $teal;
+      }
+    }
```
Comment (jasoncypret) [inline app/pb_kits/playbook/packs/site_styles/docs/_kit_doc.scss:44]:
```suggestion
    &.dark {
      color: $text_dk_light;
      a {
        color: $teal;
      }
    }
```

File: app/pb_kits/playbook/packs/site_styles/docs/_kit_doc.scss [hunk 8]
```diff
@@ -134,56 +127,6 @@
...
-      position: relative;
-      font-weight: 400;
-
-      &:after {
-        content: "not set";
-      }
-    }
-  }
+pre.highlight{
+  padding:$space_xl;
```
Comment (jasoncypret) [inline app/pb_kits/playbook/packs/site_styles/docs/_kit_doc.scss:131]:
```suggestion
pre.highlight {
  padding: $space_xl;
```

File: app/pb_kits/playbook/packs/site_styles/docs/_markdown.scss [hunk 9]
```diff
@@ -77,6 +79,10 @@
       // padding-bottom: 0;
     }
   }
-
-
+  &.dark {
+    color: $text_dk_light ;
+    h1,h2,h3,h4,h5,h6{
```
Comment (jasoncypret) [inline app/pb_kits/playbook/packs/site_styles/docs/_markdown.scss:84]:
```suggestion
    h1,h2,h3,h4,h5,h6 {
```

**Review by jasoncypret — CHANGES_REQUESTED** [review]

**Comment by thestephenmarshall** [comment]
@kre8sions somehow the branch is out of date. I think it might need `6.7.0` merged in!

**Pull request merged**